language/proto/known_go_imports.go

```diff
@@ -7,16 +7,16 @@ import "github.com/bazelbuild/bazel-gazelle/label"
 
 var knownGoProtoImports = map[string]label.Label{
 
-	"github.com/golang/protobuf/ptypes/any":               label.Label{Repo: "io_bazel_rules_go", Pkg: "proto/wkt", Name: "any_go_proto"},
-	"google.golang.org/genproto/protobuf/api":             label.Label{Repo: "io_bazel_rules_go", Pkg: "proto/wkt", Name: "api_go_proto"},
-	"github.com/golang/protobuf/protoc-gen-go/plugin":     label.Label{Repo: "io_bazel_rules_go", Pkg: "proto/wkt", Name: "compiler_plugin_go_proto"},
-	"github.com/golang/protobuf/protoc-gen-go/descriptor": label.Label{Repo: "io_bazel_rules_go", Pkg: "proto/wkt", Name: "descriptor_go_proto"},
-	"github.com/golang/protobuf/ptypes/duration":          label.Label{Repo: "io_bazel_rules_go", Pkg: "proto/wkt", Name: "duration_go_proto"},
-	"github.com/golang/protobuf/ptypes/empty":             label.Label{Repo: "io_bazel_rules_go", Pkg: "proto/wkt", Name: "empty_go_proto"},
-	"google.golang.org/genproto/protobuf/field_mask":      label.Label{Repo: "io_bazel_rules_go", Pkg: "proto/wkt", Name: "field_mask_go_proto"},
-	"google.golang.org/genproto/protobuf/source_context":  label.Label{Repo: "io_bazel_rules_go", Pkg: "proto/wkt", Name: "source_context_go_proto"},
-	"github.com/golang/protobuf/ptypes/struct":            label.Label{Repo: "io_bazel_rules_go", Pkg: "proto/wkt", Name: "struct_go_proto"},
-	"github.com/golang/protobuf/ptypes/timestamp":         label.Label{Repo: "io_bazel_rules_go", Pkg: "proto/wkt", Name: "timestamp_go_proto"},
-	"google.golang.org/genproto/protobuf/ptype":           label.Label{Repo: "io_bazel_rules_go", Pkg: "proto/wkt", Name: "type_go_proto"},
-	"github.com/golang/protobuf/ptypes/wrappers":          label.Label{Repo: "io_bazel_rules_go", Pkg: "proto/wkt", Name: "wrappers_go_proto"},
+	"github.com/golang/protobuf/ptypes/any":               label.Label{Repo: "com_github_golang_protobuf", Pkg: "ptypes/any", Name: "any"},
```

Comment:
This hard-coded resolution looks pretty tautological now - did you test what happens if it is removed?

Reply:
I tried. Two issues with that. First, these proto are like "standard libraries". If we stop hard coding them, then every repo with proto files will have to have something below:

```
# gazelle:resolve proto google/protobuf/timestamp.proto @com_google_protobuf//:timestamp_proto
# gazelle:resolve proto google/protobuf/descriptor.proto @com_google_protobuf//:descriptor_proto
# gazelle:resolve proto google/protobuf/duration.proto @com_google_protobuf//:duration_proto
# gazelle:resolve proto google/protobuf/wrappers.proto @com_google_protobuf//:wrappers_proto
# gazelle:resolve proto google/protobuf/any.proto @com_google_protobuf//:any_proto
# gazelle:resolve proto google/protobuf/field_mask.proto @com_google_protobuf//:field_mask_proto
# gazelle:resolve proto google/protobuf/struct.proto @com_google_protobuf//:struct_proto
# gazelle:resolve proto google/protobuf/empty.proto @com_google_protobuf//:empty_proto
```

Even worse, I ran into errors like:

```
application didn't start cleanly: failed to construct reflection server: raw filedescriptor for "google/protobuf/duration.proto" provided by service "x" do not match bytes provided by service "y";
```